server/src/test/scala/cromwell/CromwellCommandLineSpec.scala

```diff
@@ -47,37 +47,114 @@ class CromwellCommandLineSpec extends FlatSpec with Matchers with BeforeAndAfter
   //  }
 
   it should "run single when supplying wdl and inputs" in {
-    val optionsLast = parser.parse(Array("run", "3step.wdl", "--inputs", "3step.inputs"), CommandLineArguments()).get
+    val threeStep = WdlAndInputs(ThreeStep)
+    val optionsLast = parser.parse(Array("run", threeStep.wdl, "--inputs", threeStep.inputs), CommandLineArguments()).get
     optionsLast.command shouldBe Some(Run)
-    optionsLast.workflowSource.get.pathAsString shouldBe "3step.wdl"
-    optionsLast.workflowInputs.get.pathAsString shouldBe "3step.inputs"
+    optionsLast.workflowSource.get shouldBe threeStep.wdl
+    optionsLast.workflowInputs.get.pathAsString shouldBe threeStep.inputs
 
-    val optionsFirst = parser.parse(Array("run", "--inputs", "3step.inputs", "3step.wdl"), CommandLineArguments()).get
+    val optionsFirst = parser.parse(Array("run", "--inputs", threeStep.inputs, threeStep.wdl), CommandLineArguments()).get
     optionsFirst.command shouldBe Some(Run)
-    optionsFirst.workflowSource.get.pathAsString shouldBe "3step.wdl"
-    optionsFirst.workflowInputs.get.pathAsString shouldBe "3step.inputs"
+    optionsFirst.workflowSource.get shouldBe threeStep.wdl
+    optionsFirst.workflowInputs.get.pathAsString shouldBe threeStep.inputs
+
+    val validation = Try(CromwellEntryPoint.validateRunArguments(optionsFirst))
+    validation.isSuccess shouldBe true
+    validation.get.workflowSource shouldBe None
+    validation.get.workflowUrl shouldBe Some(threeStep.wdl)
+  }
+
+  it should "run single when supplying workflow using url" in {
```

Comment:
what does "run single" mean?

Reply:
Run single workflow/submission. The test is meant for `Run` mode.